celery/app/task.py

```diff
@@ -714,6 +714,8 @@ def retry(self, args=None, kwargs=None, exc=None, throw=True,
         if is_eager:
             # if task was executed eagerly using apply(),
             # then the retry must also be executed eagerly in apply method
+            # This is needed for eaged 
+            S.apply()
```

Comment:
@auvipy this is the fixed regression

Reply:
can you also check the related unit test?

Reply:
My problem is that i dosn't test things directly on unit test/git soulution, but debugging my "work code" entering in celery. I have to setup the env to test/make unit tests